### PR TITLE
Issue-109 Use HTL comments

### LINKED ISSUE #109: Use HTL comments
If you want to place comments regarding your code, make sure they don't display to end users.
```
<!--/* Good, this comment will not be rendered */-->
 
<!-- Potentially bad, plain HTML comments will be visible on the page -->
```

### PR DESCRIPTION
**Pull request by GulderBone**
fixes #109.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,9 @@ Below you will find descriptions of all rules available in **AEM Rules for Sonar
 - **HTL-4** Name and re-use Repeating Conditions
   - Consider caching data-sly-test conditions and reduce code duplication.
 
+- **HTL-5** Usage of HTML comments should be avoided if possible
```
Comment (piotr-wilczynski) [inline README.md]: HTML or HTL?

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+)
+public class HtlCommentsCheck extends AbstractHtlCheck {
+
+  static final String RULE_KEY = "HTL-5";
+
+  static final String RULE_MESSAGE = "Always use HTL style of comments";
+
+  @Override
+  public void comment(CommentNode node) {
+    if (!isHTLComment(node)) {
```
Comment (piotr-wilczynski) [inline src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: org.apache.sling.scripting.sightly.impl.compiler.Syntax.isSightlyComment(node.getCode())

File: src/test/files/checks/htl/HtlAttributesShouldBeAtTheEndCheck.html [hunk 3]
```diff
@@ -1,22 +1 @@
-<!--
```
Comment (piotr-wilczynski) [inline src/test/files/checks/htl/HtlAttributesShouldBeAtTheEndCheck.html]: ?

File: src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java [hunk 4]
```diff
@@ -51,7 +51,7 @@ public void sanity() {
         profile.define(context);
         Map<String, BuiltInActiveRule> activeRules = getActiveRulesByRuleKey(context);
 
-        assertThat(activeRules.size()).isEqualTo(4);
+        assertThat(activeRules.size()).isEqualTo(5);
```
Comment (piotr-wilczynski) [inline src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java]: please update this branch to the master before merge and update those values

File: README.md (hunk 1 above)
Comment (GulderBone) [reply README.md]: HTML :)

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java (hunk 2 above)
Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Nice, thanks :).

File: src/test/files/checks/htl/HtlAttributesShouldBeAtTheEndCheck.html (hunk 3 above)
Comment (GulderBone) [reply src/test/files/checks/htl/HtlAttributesShouldBeAtTheEndCheck.html]: Reverted and updated license.

File: src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java (hunk 4 above)
Comment (GulderBone) [reply src/test/java/com/cognifide/aemrules/htl/HtlProfileTest.java]: It is synced with master, and at the moment this test is passing with '5'.

File: pom.xml [hunk 5]
```diff
@@ -343,6 +343,7 @@
             <exclude>**/LICENSE.txt</exclude>
             <exclude>**.xml</exclude>
             <exclude>assets/**</exclude>
+            <exclude>src/test/files/checks/htl/HtlCommentsCheck.html</exclude>
```
Comment (jplucinski) [inline pom.xml]: This looks strange, you can skip updating licence with param: `-Dlicense.skip=true`

File: src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    technicalDebt = "5min"
+)
+public class HtlCommentsCheck extends AbstractHtlCheck {
+
+  static final String RULE_KEY = "HTL-5";
+
+  static final String RULE_MESSAGE = "Always use HTL style of comments";
+
+  @Override
+  public void comment(CommentNode node) {
```
Comment (jplucinski) [inline src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Please check if this works correctly with rule **HTL-14** - examples from `DisplayContextInConditionalCommentsCheck.html` file.

Comment (GulderBone) [reply src/main/java/com/cognifide/aemrules/htl/checks/HtlCommentsCheck.java]: Checked, added comments with conditions handling.

**Pull request merged**